src/services/aws_macros.ts

```diff
@@ -79,17 +81,19 @@ export class AWS {
   codeBuildRetryStrategy: StandardRetryStrategy;
 
   constructor(config: AWSConfig) {
-    // declare an specific slow retry strategy, to reuse in slow apis
+    // declare a specific slow retry strategy, to reuse in slow apis
     this.slowRetryStrategy = new StandardRetryStrategy(async () => SLOW_STRATEGY_RETRIES, {
       retryDecider: error => {
         // copied the default behavior from aws-sdk to fix the problem caused by Jest on isRetryableByTrait
         if (!error) {
           return false;
         }
-        if (!error.message.includes('AWS SDK error wrapper'))
-          // default behavior when running outside Jest
-          return defaultRetryDecider(error);
-        else return true; // jest has messed the error object
+        if (IASQL_ENV === 'test' && error.message.includes('AWS SDK error wrapper')) {
```

Comment:
No, please don't hook on the `IASQL_ENV` value directly. Create a new `overrideAwsRetryDecider` (or something like that) boolean that you set in `src/config/test.ts` to `true` and `false` elsewhere. That way if there is another environment in the future that needs this behavior, it can also set it.

Reply:
good solution